File: src/cucumber/java/io/tripled/social/client/CucumberInput.java
package io.tripled.social.client;

import io.tripled.social.client.presentation.Input;

import java.util.Arrays;
import java.util.Deque;
import java.util.LinkedList;
import java.util.concurrent.BlockingDeque;
import java.util.concurrent.LinkedBlockingDeque;

class CucumberInput implements Input {

  private final BlockingDeque<String> inputs;

  CucumberInput() {
    inputs = new LinkedBlockingDeque<>();
  }

  void sendInput(String input) {
    this.inputs.offer(input);
  }

  @Override
  public String read() {
    return inputs.poll();
  }
}

File: src/test/kotlin/io/tripled/social/client/acceptance/CucumberInput.kt
package io.tripled.social.client.acceptance

import io.tripled.social.client.presentation.Input
import java.util.concurrent.BlockingDeque
import java.util.concurrent.LinkedBlockingDeque

internal class CucumberInput : Input {
    private val inputs: BlockingDeque<String> = LinkedBlockingDeque()

    fun sendInput(input: String) {
        this.inputs.offer(input)
    }

    override fun read(): String? {
        return inputs.poll()
    }
}
